### PR TITLE
Adding hackbench test to avocado[V3]

### PR DESCRIPTION
**Pull request by PraveenPenguin**
Signed-off-by: praveen@linux.vnet.ibm.com
added hackbench Test case
v1 #76
v2 #116 

```
v3:rectified variable names 
v3:remove redundant variable
v3: change compare condition
```

### CODE REVIEW

File: perf/hackbench.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        perf_json = {}
+        for line in self.results.split('\n'):
+            if line.startswith('Time:'):
+                time_spent = line.split()[1]
+                perf_json = {'time': time_spent}
+        output_path = os.path.join(self.outputdir, "perf.json")
+        json.dump(perf_json, open(output_path, "w"))
+        self.log.info("Time Taken:" + time_spent)
+        if self._threshold_time:
+            if self._threshold_time >= time_spent:
```
Comment (apahim) [inline perf/hackbench.py:78]: Still failing to understand this. Shouldn't it be opposite? I mean, the test fail on `time_spent >= threshold`?

File: perf/hackbench.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                self.error("Test failed: Time Taken not "
+                           "matched with threshold")
```
Comment (apahim) [inline perf/hackbench.py:80]: Message needs update.

File: perf/hackbench.py (hunk 1 above)
Comment (PraveenPenguin) [inline perf/hackbench.py:78]: @apahim  not sure may be what default time i given in yaml file that is not suitable  for your environment , i think user need to edit every time yaml file with threshold time . 
what you think can we remove this check and just go with default run and user need to check after run as i suggested earlier ?

Comment (apahim) [inline perf/hackbench.py:78]: Sure. Your test, you decide. The threshold was only a suggestion.

Comment (PraveenPenguin) [inline perf/hackbench.py:78]: thanks , will try  once with x86 and power system  with all distribution and try to get higher threshold if there are not major gap , if not will remove this check .

Regards
Praveen

**Comment by PraveenPenguin** [comment]
@apahim  pushed changes in new PR #122 , please have a look.

Thanks

**Pull request closed without merging**